utils/roles/verify_intel_gaudi/tasks/hccl_validation.yml

```diff
@@ -0,0 +1,164 @@
+# Copyright 2024 Intel Corporation.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+---
+
+- name: Checking for the right amount of HPU devices
+  ansible.builtin.shell: |
+    set -o pipefail
+    lspci | grep "{{ verify_intel_gaudi_device_pattern }}" | wc -l
+  args:
+    executable: /bin/bash
+  register: hpu_dev
+  failed_when: ( hpu_dev.stdout != "8" )
+  changed_when: false
+
+- name: Setting python version used for the test runs
+  ansible.builtin.set_fact:
+    pver: "{{ '.'.join(ansible_python_version.split('.')[0:2]) }}"
+
+- name: Create test folder
+  ansible.builtin.tempfile:
+    state: directory
+    suffix: omnia_gaudi_hccl_test
+  register: test_folder
+
+- name: Get hccl_demo git repo tarball
+  ansible.builtin.get_url:
+    url: "{{ offline_git_path }}/hccl_demo.tar.gz"
```

Comment:
offline_git_path can get using similar tasks in https://github.com/dell/omnia/blob/release_1.7/scheduler/roles/cluster_validation/tasks/fetch_software_config.yml#L107